### DISCUSSION TITLE
predict_proba list of dicts

### DISCUSSION THREAD

**Discussion by statspy-ml** [online-ml/river#763] [Q&A] [answered]
Hello,
I am having a bad time trying to get the values from the output of predict_proba.  In scikit I just [:,1] to pass the class 1 probability do a dataframe.

How can I do this in river?

Thanks

**Comment by MaxHalford** [comment, marked as answer]
Hello 🤘

>  In scikit

scikit-learn makes use of numpy arrays and pandas DataFrames. In River, you're working with dictionaries.

It's not 100% clear to me what is your issue. The `predict_proba` method produces a dictionary that contains class probabilities. Isn't that clear to you?

You also mention "list of dicts". Are you passing a list of `predict_proba`? If you are, then you shouldn't. You should instead loop over the list and pass each individual dictionary to `predict_proba`.

*Meta: next time you open issue, I suggest providing a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). It will help us help you.*

**Reply by statspy-ml** [reply]
Hello,

Here are an example:

I guess I'm doing wrong appeding every predict_proba to a list right?
```
n_samples = 0

max_samples = 151390
predicoes =[]
proba = []  
```


```
while n_samples < max_samples:

    X, y = next(X_y)
    y_pred_proba = model.predict_proba_one(X)
    y_pred = model.predict_one(X)
    
    model.learn_one(X,y)
    predicoes.append(y_pred)
    proba.append(y_pred_proba)

    n_samples += 1
```

```
type(proba)
list
```


Now I want to pass to a column in a pandas df only the `class 1`  probabilities.

How ca I do that?


update:
I did:

```
a = y_pred_proba['1']
    proba.append(a)
```

and now I get a list that I can use the way I want.  Maybe not the smartest way, but works tho ;-)



```
proba

[{},
 {'1': 1.0},
 {'1': 1.0},
 {'1': 0.6666666666666666, '0': 0.3333333333333333},
 {'1': 0.3333333333333341, '0': 0.6666666666666671},
 {'1': 0.05745386138151764, '0': 0.9425461386184835},
 {'1': 0.9688903635304982, '0': 0.031109636469503106},
 {'1': 0.8719273922199924, '0': 0.12807260778000798},
 {'1': 0.25, '0': 0.75},
 {'1': 0.2222222222222222, '0': 0.7777777777777777},
 {'1': 0.2, '0': 0.8},
 {'1': 0.18181818181818182, '0': 0.8181818181818182},
 {'1': 0.16666666666666666, '0': 0.8333333333333333},
 {'1': 0.15384615384615385, '0': 0.8461538461538463},
 {'1': 0.14285714285714285, '0': 0.8571428571428571},
 {'1': 0.13333333333333333, '0': 0.8666666666666667},
```

**Reply by MaxHalford** [reply]
Thanks for sharing. Do you have a remaining question? What's bothering you?

**Reply by ghost** [reply]
Hi,

I am also struggling with that! How can i do the inference with a 3d tensor? 

https://github.com/online-ml/river/discussions/990

Thanks
